### ISSUE TITLE
::Trace.record throws "not implemented"

### ISSUE THREAD

**Issue by sullerandras** [openzipkin/zipkin-ruby#102]
I try to use the "annotate_plugin" method to add the status code as a binary annotation.
I created a new rails application, added the "zipkin-tracer" gem to gemfile then copied the annotate_plugin setup from the readme:

```ruby
module TestZipkinRuby
  class Application < Rails::Application
    # Settings in config/environments/* take precedence over those specified here.
    # Application configuration should go into files in config/initializers
    # -- all .rb files in that directory are automatically loaded.
    require 'zipkin-tracer'
    zipkin_config = {
      service_name: 'example-app',
      service_port: 443,
      sample_rate: 1,
      json_api_host: 'http://localhost:9411',
      annotate_plugin: lambda do |env, status, response_headers, response_body|
        ep = ::Trace.default_endpoint
        # string annotation
        ::Trace.record(::Trace::BinaryAnnotation.new('http.referrer', env['HTTP_REFERRER'], 'STRING', ep)) # this is line 23
        # integer annotation
        ::Trace.record(::Trace::BinaryAnnotation.new('http.content_size', [env['CONTENT_SIZE']].pack('N'), 'I32', ep))
        ::Trace.record(::Trace::BinaryAnnotation.new('http.status', [status.to_i].pack('n'), 'I16', ep))
      end
    }
    middleware.use ZipkinTracer::RackHandler, zipkin_config
  end
end
```

This throws `not implemented` error at runtime with this backtrace:

```
finagle-thrift (1.4.2) lib/finagle-thrift/tracer.rb:4:in `record'
finagle-thrift (1.4.2) lib/finagle-thrift/trace.rb:43:in `record'
config/application.rb:23:in `block in <class:Application>'
zipkin-tracer (0.21.1) lib/zipkin-tracer/rack/zipkin-tracer.rb:38:in `annotate_plugin'
zipkin-tracer (0.21.1) lib/zipkin-tracer/rack/zipkin-tracer.rb:48:in `trace!'
zipkin-tracer (0.21.1) lib/zipkin-tracer/rack/zipkin-tracer.rb:25:in `block (2 levels) in call'
zipkin-tracer (0.21.1) lib/zipkin-tracer/zipkin_tracer_base.rb:19:in `with_new_span'
zipkin-tracer (0.21.1) lib/zipkin-tracer/rack/zipkin-tracer.rb:24:in `block in call'
zipkin-tracer (0.21.1) lib/zipkin-tracer/trace_container.rb:8:in `with_trace_id'
zipkin-tracer (0.21.1) lib/zipkin-tracer/rack/zipkin-tracer.rb:20:in `call'
```

Is the readme out of date? Do i have to configure something else?

Ruby 2.3.3, Rails 4.28 or 5.0.2, zipkin-tracer 0.21.1

**Comment by jfeltesse-mdsol** [comment]
I believe the README is outdated, a quick grep of the code hints that `record` is now to be called on `Trace::Span`.
I know next to nothing about the annotate plugin so I can't do much more here but diving in the code shouldn't be that hard, this gem is actually pretty compact.

FYI @jcarres-mdsol

**Comment by strom** [comment]
It looks like the error above is calling this `record`: https://github.com/twitter/finagle/blob/develop/finagle-thrift/src/main/ruby/lib/finagle-thrift/tracer.rb#L3

**Comment by strom** [comment]
This could be fixed if the `annotate_plugin` function passes in the span as an argument. That way, you could have the span available in the lambda passed to the plugin and you can do `span.record`.

**Comment by jcarres-mdsol** [comment]
Addressed by #106 . Thanks @elyara !
